lib/blocktree/node.go

```diff
@@ -112,7 +112,7 @@ func (n *node) getLeaves(leaves []*node) []*node {
 		leaves = []*node{}
 	}
 
-	if n.children == nil || len(n.children) == 0 {
```

Comment:
couldnt removing this nil check cause a panic in the following condition?

Reply:
Calling `len()` on a `nil` slice returns 0.